azkaban-common/src/main/java/azkaban/jobExecutor/AbstractProcessJob.java

```diff
@@ -161,15 +176,21 @@ public Props getJobGeneratedProperties() {
    * @return {tmpPropFile, outputPropFile}
    */
   public File[] initPropsFiles() {
-    // Create properties file with additionally all input generated properties.
-    final File[] files = new File[2];
-    files[0] = createFlattenedPropsFile(this._cwd);
+    // Create job properties file with additionally all input generated properties.
+    final File[] files = new File[3];
+    files[0] = createFlattenedPropsFile(this.jobProps, this._cwd, "_job_props_");
 
     this.jobProps.put(ENV_PREFIX + JOB_PROP_ENV, files[0].getAbsolutePath());
     this.jobProps.put(ENV_PREFIX + JOB_NAME_ENV, getId());
 
     files[1] = createOutputPropsFile(getId(), this._cwd);
     this.jobProps.put(ENV_PREFIX + JOB_OUTPUT_PROP_FILE, files[1].getAbsolutePath());
+
+    // Create job's private properties file.
```

Comment:
Why do we need to add jobtype plugin private properties to a file in the execution directory?